File: persistencia/src/main/java/co/edu/uniquindio/proyecto/dto/UsuarioGetDTO.java
package co.edu.uniquindio.proyecto.dto;

import co.edu.uniquindio.proyecto.entidades.*;
import lombok.*;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class UsuarioGetDTO implements Serializable {
    private String cedula;
    private String nombre;
    private String email;
    private List<ProductoGetDTO> productos = new ArrayList<>(0);
    private List<ProductoGetDTO> productosFavoritos = new ArrayList<>(0);
    private CarritoGetDTO carrito;
    private Boolean isCuentaActiva;
    private List<ComentarioGetDTO> comentarios = new ArrayList<>(0);
    private List<TransaccionGetDTO> compras = new ArrayList<>(0);
    private List<TransaccionGetDTO> ventas = new ArrayList<>(0);
    private String telefono;
    private String direccion;

}

File: my-app/src/app/modelo/usuario-get-dto.ts
import { CarritoGetDTO } from "./carrito-get-dto";
import { ComentarioGetDTO } from "./comentario-get-dto";
import { ProductoGetDTO } from "./producto-get-dto";
import { TransaccionGetDTO } from "./transaccion-get-dto";

export class UsuarioGetDTO {
cedula: string = "";
nombre: string = "";
email: string = "";
productos: ProductoGetDTO[] = [];
productosFavoritos: ProductoGetDTO[] = [];
carrito: CarritoGetDTO ;
isCuentaActiva: boolean = true;
comentarios: ComentarioGetDTO[] = [];
compras: TransaccionGetDTO[] = [];
ventas: TransaccionGetDTO[] = [];
telefono: string = "";
direccion: string = "";


constructor(
    carrito: CarritoGetDTO,
    cedula: string,
    nombre: string,
    email: string,
    productos: ProductoGetDTO[],
    productosFavoritos: ProductoGetDTO[],
    isCuentaActiva: boolean,
    comentarios: ComentarioGetDTO[] = [],
    compras: TransaccionGetDTO[] = [],
    ventas: TransaccionGetDTO[] = [],
    telefono: string = "",
    direccion: string = ""
  ) {
    this.carrito = carrito;
    this.cedula = cedula;
    this.nombre = nombre;
    this.email = email;
    this.productos = productos;
    this.productosFavoritos = productosFavoritos;
    this.isCuentaActiva = isCuentaActiva;
    this.comentarios = comentarios;
    this.compras = compras;
    this.ventas = ventas;
    this.telefono = telefono;
    this.direccion = direccion;
  }
}
